### DISCUSSION TITLE
create a sub page with the buildnumber

### DISCUSSION THREAD

**Discussion by brux88** [rfennell/AzurePipelines#1234] [Q&A]
hello congratulations for the beautiful project

is it possible to create a zip asset of the update to download and put the link directly in the release notes?

**Comment by rfennell** [comment]
If you publish an Azure DevOps pipeline artifacts you could access the list via a link from the release notes. 

```
<a href="{{builddetails.web.href}}&view=artifacts&type=publishedArtifacts">A build artifacts</a>
```

You can use the same basic process assuming you have a well know location for the ZIP file so you can build the URL

**Reply by brux88** [reply]
I have my zip file in this path in the pipiline when I archive:
'$ (Build.ArtifactStagingDirectory) /Release.zip'


how do i use this path in the href?

**Reply by rfennell** [reply]
The zip file won't be in the staging directory once the pipeline ends, that folder is a working folder. 

You need to publish the contents of the $ (Build.ArtifactStagingDirectory)  folder somewhere such as pipeline artifact (see https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/utility/publish-pipeline-artifact?view=azure-devops) or maybe to an Azure DevOps Artifact Feed (see 
https://docs.microsoft.com/en-us/azure/devops/artifacts/concepts/feeds?view=azure-devops)

As long as you know the URL of where you publish to, you can create a HREF to it as I mentioned above

**Comment by brux88** [comment]
that's right, after I create the zip file, then I publish it as an artifact:

  - task: PublishBuildArtifacts @ 1
    inputs:
      PathtoPublish: '$ (Build.ArtifactStagingDirectory)'
      ArtifactName: 'drop'
      publishLocation: 'Container'
      
      
      
      then in the release tasks I find the zip in the artifact download, the photo is the agent job setting
![2022-02-10_12-00-58](https://user-images.githubusercontent.com/3495339/153393828-75d3c260-c9b5-4b94-b755-71373ce64d21.png)

**Reply by rfennell** [reply]
I can see your Classic Release agent stage is consuming the `_sedna.zip` artefact that your linked build must have created.

The JSON object created by my task that is passed into the Handlebars template, when used in a Classic Release, contains an array `builds` in which all the artifacts referenced in the Release are listed. You can use this array to get the base URL to the build and hence build its published artifacts URL. 

As you appears to only have a single build artifact you can assume the template will be something like

```
<a href="{{builds[0].build._links.web.href}}&view=artifacts&type=publishedArtifacts">Link to ZIP build artifacts</a>
```

To help with this process and to see what you have to work worth you can 

- Look at dumping out the `payload.json` so you can see the data you have available using the settings

```
    dumpPayloadToFile: true
    dumpPayloadFileName: <some path you can access>payload.json
```

- Or use the [local tester I provide](https://github.com/rfennell/AzurePipelines/blob/main/Extensions/XplatGenerateReleaseNotes/V3/testconsole/readme.md), this is tool I use to work out the template I need, it allows easy experimentation
